presto-main/src/main/java/com/facebook/presto/operator/scalar/IpPrefixFunctions.java

```diff
@@ -197,6 +197,54 @@ public static Block collapseIpPrefixes(@SqlType("array(IPPREFIX)") Block unsorte
         return blockBuilder.build();
     }
 
+    @Description("Split the input prefix into subnets the size of the new prefix length.")
+    @ScalarFunction("ip_prefix_subnets")
+    @SqlType("array(IPPREFIX)")
+    public static Block ipPrefixSubnets(@SqlType(StandardTypes.IPPREFIX) Slice prefix, @SqlType(StandardTypes.BIGINT) long newPrefixLength)
+    {
+        boolean inputIsIpV4 = isIpv4(prefix);
+
+        if (newPrefixLength < 0 || (inputIsIpV4 && newPrefixLength > 32) || (!inputIsIpV4 && newPrefixLength > 128)) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Invalid prefix length for IPv" + (inputIsIpV4 ? "4" : "6") + ": " + newPrefixLength);
+        }
+
+        int inputPrefixLength = getPrefixLength(prefix);
+        // We can only disaggregate a prefix if the prefix length is longer (more-specific) than the prefix length of the input prefix.
+        // E.g., if the input prefix is /24, the subnetSize must be like /25, /26 etc....
```

Comment:
Use the full prefix string here. e.g. 9.255.255.0/24, or use the word "prefix length" and explain that the prefix is composed of IP/(prefix lengtgh).